File: Hackerrank/30days/cpp/day20.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n;
    cin >> n;
    vector<int> a(n);
    for(int a_i = 0; a_i < n; a_i++){
    	cin >> a[a_i];
    }
    // Write Your Code Here
    int numSwaps = 0;
    for (int i = 0; i < n; i++){

        for (int j = 0; j < n - i- 1; j++){

            if (a[j] > a[j + 1]){

                int temp = a[j];
                a[j] = a[j + 1];
                a[j + 1] = temp;
                numSwaps++;

            }
        }
    }
    cout << "Array is sorted in " << numSwaps << " swaps." << endl;
    cout << "First Element: " << a[0] << endl;
    cout << "Last Element: " << a[n-1];
    return 0;
}

File: Hackerrank/30days/day20.py
import sys

n = int(input().strip())
a = list(map(int, input().strip().split(' ')))
# Write Your Code Here

numSwaps = 0
for i in range(n - 1):
    for j in range(n-i-1):
        if (a[j] > a[j + 1]):
            a[j], a[j + 1] = a[j + 1], a[j]
            numSwaps += 1
    if numSwaps == 0:
        break


print('Array is sorted in {} swaps.'.format(numSwaps))
print('First Element: {}'.format(a[0]))
print('Last Element: {}'.format(a[-1]))